pkg/cfg/sanitizer.go

```diff
@@ -0,0 +1,47 @@
+package cfg
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+type Sanitizer func(in interface{}) (interface{}, error)
+
+func Sanitize(in map[string]interface{}, sanitizers []Sanitizer) (interface{}, error) {
+	var err error
+
+	if len(sanitizers) == 0 {
+		return in, nil
+	}
+
+	for key, val := range in {
+		for _, san := range sanitizers {
+			if in[key], err = san(val); err != nil {
+				return nil, fmt.Errorf("can not apply sanitizer on key %s: %w", key, err)
+			}
+		}
+
+		if valMap, ok := val.(map[string]interface{}); ok {
```

Comment:
Will this code be actually reached? When I understand it correctly, any error (of course depending on the sanitizer), will trigger a return from this function.. Thus giving a map as a parameter to all the sanitizers might return prematurely.